do_mpc/controller.py

```diff
@@ -1315,8 +1350,12 @@ def _create_nlp(self):
             'expand': False,
             'ipopt.linear_solver': 'mumps',
         }.update(self.nlpsol_opts)
+        # Is it a MINLP?
+        MINLP = {"discrete":self.integer} if True in self.integer else self.nlpsol_opts
```

Comment:
This I don't really like. Please don't rename the ``self.nlpsol_opts``. I would much prefer something like:

```python 
# Check if MINLP: Update nlpsol_opts and set solvername accordingly
if np.any(self.opt_x_integer_flag.cat):
   self.nlpsol_opts.update({
       'discrete': self.opt_x_integer_flag
       })
   solvername = 'bonmin'
else:
   solvername = 'ipopt'

self.S = nlpsol('S', solvername, nlp, self.nlpsol_opts)


```